### ISSUE TITLE
Create an embed widget link

### ISSUE THREAD

**Issue by gubuntu** [kartoza/geonode#153]
Link that gives you an embed widget for embedding a map in your own site. This is another variant of the above approaches, but this time will return an embeddable widget as html and JavaScript snippets that can be incorporated into an existing web site. Implemented for QGIS backend only.

child of #457

**Comment by timlinux** [comment]
Please test, this does not work for me - see #336

**Comment by gubuntu** [comment]
broken
![Screen Shot 2017-10-20 at 18.23.58.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/53ba46167953d2fd0f723be0/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c57e0b76236456f36b7191a379e70c2952415a46be5d4a1f06a3a3dc78e2a436628f595fef84645d70c43e44a4bcd41fe901ba73de4bd0b7b89850d2aa6c7bb3d4d216744b470a83ed5c32aa984ec850f34ae74b5da7113363f43b.png)

**Comment by boney-bun** [comment]
#376 should fix the bug on firefox

**Comment by boney-bun** [comment]
it's fixed on the testing.
could we close this issue?

**Comment by gubuntu** [comment]
The embed map code does get generated. But please actually test it by embedding it in a page and confirming that it works before we close this.

**Comment by NyakudyaA** [comment]
@boney-bun  Tested this and it's still not rendering the actual layers. The extent is correct but the layer is not rendering:
![screenshot from 2018-02-15 11-39-01](https://user-images.githubusercontent.com/2510900/36249979-1c6f7cf2-1245-11e8-89df-60c3142ee00c.png)

**Comment by NyakudyaA** [comment]
@boney-bun  Could we also have the same icon layout for the widget links.
This one has an option for a user to copy the code.
![embed_widget_link](https://user-images.githubusercontent.com/2510900/36250344-31092f54-1246-11e8-8133-69082a3a5296.png)

Add the copy code link as well:
![embed_iframe_link](https://user-images.githubusercontent.com/2510900/36250400-53e14052-1246-11e8-888c-1b85ce1a3e69.png)

**Comment by boney-bun** [comment]
from #394:
## Steps on create an embed widget link
clicks embed map on a map. 
Then choose embed iframe link.
copy the snippet on a text editor, save as a html file.
loads the html file in the browser.
the result is all layers will be rendered instead, not specifically for the chosen map.

## Expected Result
The html file should show only the layers within the map, not all layers.

**Comment by boney-bun** [comment]
> @boney-bun Could we also have the same icon layout for the widget links.

@NyakudyaA #410 addresses your sugggestion.

**Comment by gubuntu** [comment]
@boney-bun please post a screencast to show an embed widget working before we can close this.

**Comment by boney-bun** [comment]
@gubuntu here are the screencast:
using firefox:
![geonode 153_2](https://user-images.githubusercontent.com/4602383/42992266-3ecb1844-8c32-11e8-9dad-715589c7102b.gif)

using chrome:
![geonode 153](https://user-images.githubusercontent.com/4602383/42992267-3f039156-8c32-11e8-9882-4f9b0b861de1.gif)

**Comment by NyakudyaA** [comment]
TEsted widget link in Chrome,Firefox and opera mini and it all works. I will close this and open another issue that relates to the embeded iframe link
